commit df8af5b0953791cb27c73a7f67b8101fedab4ca0
Author: William Lallemand <wlallemand@haproxy.com>
Date:   Tue Jan 14 15:38:43 2020 +0100

    REGTEST: mcli/mcli_start_progs: start 2 programs
    
    This regtest tests the issue #446 by starting 2 programs and checking if
    they exist in the "show proc" of the master CLI.
    
    Should be backported as far as 2.0.
    
    (cherry picked from commit 25b569302167e71b32e569a2366027e8e320e80a)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/reg-tests/mcli/mcli_start_progs.vtc b/reg-tests/mcli/mcli_start_progs.vtc
new file mode 100644
index 000000000..bda9b9601
--- /dev/null
+++ b/reg-tests/mcli/mcli_start_progs.vtc
@@ -0,0 +1,40 @@
+varnishtest "Try to start a master CLI with 2 programs"
+#REGTEST_TYPE=bug
+#REQUIRE_VERSION=2.0
+#REQUIRE_BINARIES=sleep
+
+feature ignore_unknown_macro
+
+# Do nothing. Is there only to create s1_* macros
+server s1 {
+} -start
+
+haproxy h1 -W -S -conf {
+    global
+        nbproc 1
+    defaults
+        mode http
+        ${no-htx} option http-use-htx
+        timeout connect 1s
+        timeout client  1s
+        timeout server  1s
+
+    frontend myfrontend
+        bind "fd@${my_fe}"
+        default_backend test
+
+    backend test
+        server www1 ${s1_addr}:${s1_port}
+
+    program foo
+        command sleep 10
+
+    program bar
+        command sleep 10
+
+} -start
+
+haproxy h1 -mcli {
+    send "show proc"
+    expect ~ ".*foo.*\n.*bar.*\n"
+} -wait
